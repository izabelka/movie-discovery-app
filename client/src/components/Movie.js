import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import {
  getMovieDetails,
  updateAppState,
  setPrevAppState,
} from '../stores/actionCreators';

class Movie extends Component {

  onMovieClick = () => {
    if (this.props.position) {
      this.props.setPrevAppState('initial');
    } else {
      this.props.setPrevAppState('search-results');
    }
    this.props.getMovieDetails(this.props.movieId);
    this.props.updateAppState('movie-details');
  }

  render() {
    let {
      title,
      poster_path: posterPath,
      vote_average: voteAverage,
      overview,
    } = this.props.movie;
    let position = this.props.position;
    overview = overview.length > 300 ? overview.slice(0, 300) + '…' : overview;
    return (
      <Wrapper
        onClick={this.onMovieClick}>
        {position &&
          <PositionMarker>
            <Position>
              {position}
            </Position>
          </PositionMarker>
        }
        {posterPath &&
          <Poster
            src={`http://image.tmdb.org/t/p/w342//${posterPath}`} />
          }
        <MovieInfo>
          <Title>
            {title}
          </Title>
          <VoteAverage>
            {`⭐️${voteAverage}`}
          </VoteAverage>
          <Description>
            {overview}
          </Description>
        </MovieInfo>
      </Wrapper>
    );
  }
}

const Wrapper = styled.div`
  width: 100%;
  padding: 16px;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  border: 1px solid #c8d1dc;
  border-radius: 4px;
  position: relative;
  cursor: pointer;

  &:not(:last-of-type){
    margin-bottom: 22px;
  }
`;

const PositionMarker = styled.div`
  position: absolute;
  top: -16px;
  left: -16px;
  width: 44px;
  height: 44px;
  border-radius: 100px;
  background-color: yellow;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1;
`;

const Position = styled.span`
  color: #000;
  font-weight: 800;
  font-size: 2em;
`;

const Poster = styled.img`
  height: 240px;
  margin-right: 34px;
`;

const MovieInfo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
`;

const Title = styled.h2`
  font-size: 1.8em;
  font-weight: 600;
  margin-bottom: 10px;
`;

const VoteAverage = styled.span`
  font-size: 1.2em;
  line-height: 1.2em;
  vertical-align: middle;
  margin-bottom: 14px;
`;

const Description = styled.p`
  line-height: 1.4em;
  font-size: 1.3em;
`;

const mapStateToProps = state => ({
  appState: state.appState,
});

const mapDispatchToProps = dispatch => ({
 getMovieDetails: (movieId) => dispatch(getMovieDetails(movieId)),
 updateAppState: (appState) => dispatch(updateAppState(appState)),
 setPrevAppState: (appState) => dispatch(setPrevAppState(appState)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Movie);
