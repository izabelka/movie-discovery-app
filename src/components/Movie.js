import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

class Movie extends Component {
  render() {
    let {
      title,
      posterPath,
      voteAverage,
      overview,
      position,
    } = this.props;
    return (
      <Wrapper>
        <PositionMarker>
          <Position>
            {position}
          </Position>
        </PositionMarker>
        <Poster
          src={`http://image.tmdb.org/t/p/w342//${posterPath}`} />
        <MovieInfo>
          <Title>
            {title}
          </Title>
          <VoteAverage>
            {voteAverage}
          </VoteAverage>
          <Description>
            {overview}
          </Description>
        </MovieInfo>
      </Wrapper>
    );
  }
}

const Wrapper = styled.div`
  width: 100%;
  padding: 16px;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  border: 1px solid #c8d1dc;
  border-radius: 4px;
  color: #fff;
  position: relative;

  &:not(:last-of-type){
    margin-bottom: 22px;
  }
`;

const PositionMarker = styled.div`
  position: absolute;
  top: -16px;
  left: -16px;
  width: 44px;
  height: 44px;
  border-radius: 100px;
  background-color: yellow;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1;
`;

const Position = styled.span`
  color: #000;
  font-weight: 600;
  font-size: 2em;
`;

const Poster = styled.img`
  height: 240px;
  margin-right: 34px;
`;

const MovieInfo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
`;

const Title = styled.h2`
  font-size: 1.6em;
  font-weight: 600;
`;

const VoteAverage = styled.span``;

const Description = styled.p`
  line-height: 1.4em;
  font-size: 1.3em;
`;

const mapStateToProps = state => ({
  //...state,
});

const mapDispatchToProps = dispatch => ({
});

export default connect(mapStateToProps, mapDispatchToProps)(Movie);
