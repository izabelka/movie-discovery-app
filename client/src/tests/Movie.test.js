import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Movie from '../components/Movie';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from '../store';
import { Provider } from 'react-redux'

configure({ adapter: new Adapter() });

describe('Movie', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Provider store={configureStore()}>
        <Movie
          key="mockKey"
          movieId="mockMovieId"
          title="mockTitle"
          voteAverage="mockVoteAverage"
          posterPath="mockPosterPath"
          overview="mockOverview"
          position="mockPosition" />
      </Provider>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
