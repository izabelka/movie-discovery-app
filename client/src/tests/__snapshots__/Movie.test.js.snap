// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render correctly 1`] = `
<Connect(Movie)
  key="mockKey"
  movieId="mockMovieId"
  overview="mockOverview"
  position="mockPosition"
  posterPath="mockPosterPath"
  title="mockTitle"
  voteAverage="mockVoteAverage"
/>
`;
